content/style-injector.js

```diff
@@ -19,90 +19,91 @@ self.createStyleInjector = self.INJECTED === 1 ? self.createStyleInjector : ({
   let isTransitionPatched;
   // will store the original method refs because the page can override them
   let creationDoc, createElement, createElementNS;
+
   return {
-    apply,
-    clear,
-    clearOrphans,
-    remove,
-    replace,
-    toggle,
+
     list,
-  };
 
-  function apply(styleMap) {
-    const styles = _styleMapToArray(styleMap);
-    return !styles.length ?
-      Promise.resolve([]) :
-      docRootObserver.evade(() => {
-        if (!isTransitionPatched) _applyTransitionPatch(styles);
-        const els = styles.map(_apply);
-        _emitUpdate();
-        return els;
-      });
-  }
+    apply(styleMap) {
+      const styles = _styleMapToArray(styleMap);
+      return (
+        !styles.length ?
+          Promise.resolve([]) :
+          docRootObserver.evade(() => {
+            if (!isTransitionPatched && isEnabled) {
+              _applyTransitionPatch(styles);
+            }
+            return styles.map(_addUpdate);
+          })
+      ).then(_emitUpdate);
```

Comment:
Previously, _emitUpdate wasn't called on an empty list which I think was wrong. To simplify the Promise chain _emitUpdate return value is its parameter.